### DISCUSSION TITLE
Setting custom options in apple-xcframework.sh

### DISCUSSION THREAD

**Discussion by seblalayan** [jedisct1/libsodium#1335] [Q&A]
When running apple-xcframework.sh, this messages appears in the terminal:

```
Warnings related to headers being present but not usable are due to functions
that didn't exist in the specified minimum iOS version level.
They can be safely ignored.

Define the LIBSODIUM_FULL_BUILD environment variable to build the full
library (including all deprecated/undocumented/low-level functions).

Define the LIBSODIUM_SKIP_SIMULATORS environment variable to skip building
the simulators libraries (iOS, watchOS, tvOS, visionOS simulators).

Building for macOS...
```

How am I supposed to define these variables, and to what values should I assign them?

Thanks!

**Comment by jedisct1** [comment]
You can set these as environment variables (`env LIBSODIUM_FULL_BUILD=1`); their actual content doesn't matter.

But in virtually all cases, you should use normal builds, by not setting anything.

**Reply by seblalayan** [reply]
Ok so you’re telling me I can ignore this terminal message?

**Reply by seblalayan** [reply]
Ok it's fixed, for anyone struggling on a Mac:

Just install libsodium with homebrew.

`brew install libsodium`
